File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> prevMap = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            int check = target - nums[i];
            if (prevMap.containsKey(check)) {
                int[] indexes = {prevMap.get(check), i};
                return indexes;
            }
            prevMap.put(nums[i], i);
        }
        return new int[]{};
    }
}

File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        start = 0
        end = len(nums) - 1
        mid = (start + end) // 2
        while nums[mid] != target and start < end:
            if target >= nums[mid]:
                start = mid + 1
            else:
                end = mid - 1
            mid = (start + end) // 2
        if nums[mid] == target:
            return mid
        else:
            return -1

